### PR TITLE
fix:aws oss

### LINKED ISSUE #503: aws file will panic
**What happened**:

aws oss file implement will panic 

image

here can't use nil for option args 

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by akkw**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #503 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: components/file/s3/aws/oss_test.go [hunk 1]
```diff
@@ -96,6 +96,25 @@ func TestAwsOss_Put(t *testing.T) {
 
 	req.FileName = "/a.txt"
 	err = oss.Put(context.Background(), req)
-	assert.Equal(t, err.Error(), "awsoss put file[/a.txt] fail,err: invalid fileName format")
```
Comment (seeflood) [inline components/file/s3/aws/oss_test.go:99]: Why delete this assertion?

File: components/file/s3/aws/oss_test.go [hunk 2]
```diff
@@ -96,6 +96,25 @@ func TestAwsOss_Put(t *testing.T) {
...
+
+	req := &file.GetFileStu{
+		FileName: "",
+	}
+	_, err = oss.Get(context.Background(), req)
+	assert.Equal(t, err.Error(), "awsoss get file[] fail,err: invalid fileName format")
 
+	req.FileName = "layotto/get_test.txt"
+	_, err = oss.Get(context.Background(), req)
+	//assert.Nil(t, , nil)
```
Comment (seeflood) [inline components/file/s3/aws/oss_test.go:119]: Seems that we can add some assertion here

**Review by seeflood — CHANGES_REQUESTED** [review]

File: components/file/s3/aws/oss_test.go (hunk 1 above)
Comment (akkw) [reply components/file/s3/aws/oss_test.go:99]: Now this assertion can not pass without environment, can be added later

File: components/file/s3/aws/oss_test.go (hunk 2 above)
Comment (akkw) [reply components/file/s3/aws/oss_test.go:119]: Now this assertion can not pass without environment, can be added later

File: components/file/s3/aws/oss_test.go (hunk 1 above)
Comment (seeflood) [reply components/file/s3/aws/oss_test.go:99]: OK. The existing test cases depend on aws environment. We can refactor them with mocks in the future

**Review by seeflood — APPROVED** [review]
lgtm. Thanks for your contribution!

File: components/file/s3/aws/oss_test.go (hunk 1 above)
Comment (seeflood) [reply components/file/s3/aws/oss_test.go:99]: @akkw Hi, I read through the code and found that we don't have to delete this line.
The existing ut don't have dependency on remote aws server. This error is raised by https://github.com/mosn/layotto/blob/223945e4a49c499ac66a5c13b267af6b080a79a1/components/file/s3/aws/oss.go#L160

So, the s3 component get this error before connecting to the remote aws server, and we don't need an aws s3 account to run the UT

File: components/file/s3/aws/oss_test.go (hunk 2 above)
Comment (seeflood) [reply components/file/s3/aws/oss_test.go:119]: Same as above. We don't have to connect to the real server in this UT. 
Solution A: mock aws s3 client 
Solution B: simply assert that we got an error for illegal arguments.

**Review by seeflood — CHANGES_REQUESTED** [review]

File: components/file/s3/aws/oss_test.go (hunk 1 above)
Comment (akkw) [reply components/file/s3/aws/oss_test.go:99]: Yes, it is now connected to the remote service

File: components/file/s3/aws/oss_test.go (hunk 2 above)
Comment (akkw) [reply components/file/s3/aws/oss_test.go:119]: okay

**Review by seeflood — APPROVED** [review]

**Review by JervyShi — APPROVED** [review]
LGTM

**Comment by seeflood** [comment]
Thanks for your contribution!

**Pull request merged**